dashboard/app/models/pd/workshop_survey.rb

```diff
@@ -127,15 +121,66 @@ def validate_required_fields
 
     # if this is the first survey completed by this user, also require
     # demographics questions
-    if pd_enrollment && self.class.find_by_user(pd_enrollment.user).empty?
-      demographics_required_fields.each do |field|
-        add_key_error(field) unless hash.key?(field)
+    if pd_enrollment
+      if pd_enrollment.user.nil? || self.class.find_by_user(pd_enrollment.user).empty?
+        demographics_required_fields.each do |field|
+          add_key_error(field) unless hash.key?(field)
+        end
       end
     end
 
     super
   end
 
+  # Simple helper that iterates over each facilitator as reported by the user
+  # and each facilitator-specific field and yields a block with the facilitator,
+  # the field, and the combined field name we expect in the flattened version of
+  # our hash
+  def each_facilitator_field(hash=nil)
+    hash ||= sanitize_form_data_hash
+
+    facilitators = hash.try(:[], :who_facilitated) ||
+      hash.try(:[], 'whoFacilitated') ||
```

Comment:
basically because ruby-land likes hash keys to be underscored symbols and JSON-land likes object keys to be camelCased strings and I want to be able to support both.

I'd rather not normalize because I'd rather have whatever's calling this helper method be able to trust that the values returned match the hash they called it with.